inference/core/workflows/core_steps/models/foundation/segment_anything/v1.py

```diff
@@ -0,0 +1,293 @@
+        examples=["$steps.object_detection_model.predictions"],
+        default=None,
+    )
+
+    sam2_model: Union[
+        WorkflowParameterSelector(kind=[STRING_KIND]),
+        Literal["hiera_large", "hiera_small", "hiera_tiny", "hiera_b_plus"],
+    ] = Field(
+        default="hiera_large",
+        description="Model to be used.  One of hiera_large, hiera_small, hiera_tiny, hiera_b_plus",
+        examples=["hiera_large", "$inputs.openai_model"],
+    )
+
+    @classmethod
+    def accepts_batch_input(cls) -> bool:
+        return True
+
+    @classmethod
+    def describe_outputs(cls) -> List[OutputDefinition]:
+        return [
+            OutputDefinition(
+                name="predictions",
+                kind=[BATCH_OF_INSTANCE_SEGMENTATION_PREDICTION_KIND],
+            ),
+        ]
+
+    @classmethod
+    def get_execution_engine_compatibility(cls) -> Optional[str]:
+        return ">=1.0.0,<2.0.0"
+
+
+class SegmentAnything2BlockV1(WorkflowBlock):
+
+    def __init__(
+        self,
+        model_manager: ModelManager,
+        api_key: Optional[str],
+        step_execution_mode: StepExecutionMode,
+    ):
+        self._model_manager = model_manager
+        self._api_key = api_key
+        self._step_execution_mode = step_execution_mode
+
+    @classmethod
+    def get_init_parameters(cls) -> List[str]:
+        return ["model_manager", "api_key", "step_execution_mode"]
+
+    @classmethod
+    def get_manifest(cls) -> Type[WorkflowBlockManifest]:
+        return BlockManifest
+
+    def run(
+        self,
+        images: Batch[WorkflowImageData],
+        sam2_model: str,
+        boxes: Batch[sv.Detections],
+    ) -> BlockResult:
+        if self._step_execution_mode is StepExecutionMode.LOCAL:
+            return self.run_locally(images=images, sam2_model=sam2_model, boxes=boxes)
+        elif self._step_execution_mode is StepExecutionMode.REMOTE:
+            raise NotImplementedError(
+                "Remote execution is not supported for Segment Anything."
+            )
+        else:
+            raise ValueError(
+                f"Unknown step execution mode: {self._step_execution_mode}"
+            )
+
+    def run_locally(
+        self,
+        images: Batch[WorkflowImageData],
+        sam2_model: str,
+        boxes: Batch[sv.Detections],
+    ) -> BlockResult:
+
+        predictions = []
+
+        if not boxes:
+            boxes = [None] * len(images)
+
+        for single_image, boxes_for_image in zip(images, boxes):
+            prompt_class_ids = []
+            prompt_class_names = []
+            prompt_index = 0
+
+            prompts = Sam2PromptSet()
+            if boxes_for_image is not None:
+                for x1, y1, x2, y2 in boxes_for_image.xyxy:
+
+                    prompt_class_ids.append(boxes_for_image.class_id[prompt_index])
+                    prompt_class_names.append(
+                        boxes_for_image.data["class_name"][prompt_index]
```

Comment:
use constant for `class_name`

Reply:
unsure which one to use or add a new one.

I see:
 - KEYPOINTS_CLASS_NAME_KEY_IN_INFERENCE_RESPONSE here: https://github.com/roboflow/inference/blob/sam2-block/inference/core/workflows/execution_engine/constants.py#L9
 
- DetectionsPropert.CLASS_NAME here:  https://github.com/roboflow/inference/blob/sam2-block/inference/core/workflows/core_steps/common/query_language/entities/enums.py#L30  


Maybe second one (although out of scope for UQL) or add a DETECTIONS_CLASS_NAME in exec_engine.constants?

Reply:
added as `DETECTIONS_CLASS_NAME_FIELD` const at top of this file for now